tabular/src/autogluon/tabular/learner/abstract_learner.py

```diff
@@ -138,6 +138,21 @@ def predict(self, X: DataFrame, model=None, as_pandas=False):
     def _validate_fit_input(self, X: DataFrame, **kwargs):
         if self.label not in X.columns:
             raise KeyError(f"Label column '{self.label}' is missing from training data. Training data columns: {list(X.columns)}")
+        self.weight_column = kwargs.get('weight_column', None)
+        self.weight_evaluation = kwargs.get('weight_evaluation', False)
+        X_val = kwargs.get('X_val', None)
+        if self.weight_column is not None:
+            if self.weight_column not in X.columns:
+                raise KeyError(f"Weight column '{self.weight_column}' is missing from training data. Training data columns: {list(X.columns)}")
+            weight_vals = X[self.weight_column]
+            if weight_vals.isna().sum() > 0:
+                raise ValueError(f"Sample weights in column '{self.weight_column}' cannot be nan")
+            if weight_vals.dtype.kind not in 'biuf':
+                raise ValueError(f"Sample weights in column '{self.weight_column}' must be numeric values")
+            if weight_vals.min() < 0:
+                raise ValueError(f"Sample weights in column '{self.weight_column}' must be nonnegative")
+            if self.weight_evaluation and X_val is not None and self.weight_column not in X_val.columns:
```

Comment:
Does it make any sense to have sample_weight in unlabelled data?

Reply:
I don't think at this time, can add support once there is method that can exploit it.